client/normal_scalar.go

```diff
@@ -86,6 +86,14 @@ func (v normalDocument) Document() (*Document, bool) {
 	return v.val, true
 }
 
+type normalJSON struct {
+	baseNormalValue[any]
+}
+
+func (v normalJSON) JSON() (any, bool) {
```

Comment:
todo: this doesn't quite conform to the current style of normal values.

If you want to make it work as a new separate type available as normal value, you should add it to the `NormalValue` interface. And please make sure you follow the convention for regular (required) fields and optional (nillable) fields. You'd also need to include this into current tests for normal values and add new ones if needed.

All current normal scalar type can also be array elements, but I'm not sure if we want or need to enable array of jsons. An array of json can also be presented as json array after all.

It is also a bit unusual to have `any` type as underlying type for `normalJSON` which is supposed to be a strong type like all others.

Reply:
> It is also a bit unusual to have any type as underlying type for normalJSON which is supposed to be a strong type like all others.

It has to be of type `any` because the JSON value can hold any valid scalar or list of scalars. Is there an alternative to adding the `NormalJSON` type that I have missed?

Reply:
Another option could be to define a json type, this would allow us to differentiate between a known json value and an unknown `any` type:
```
type Json any // no =, not just an alias
```

Reply:
> Another option could be to define a json type, this would allow us to differentiate between a known json value and an unknown `any` type:
> 
> ```
> type Json any // no =, not just an alias
> ```

I tried this but it doesn't quite work because `Json` would also match against `any` in a type switch. I ended up having to wrap the value inside of a struct instead and implement the `Unrwap` method of the normal value.